components/script/dom/globalscope.rs

```diff
@@ -2848,6 +2849,28 @@ impl GlobalScope {
         is_js_evaluation_allowed == CheckResult::Allowed
     }
 
+    pub(crate) fn should_navigation_request_be_blocked(&self, url: &ServoUrl) -> bool {
+        let Some(csp_list) = self.get_csp_list() else {
+            return false;
+        };
+        let request = Request {
+            url: url.clone().into_url(),
+            origin: url.origin().into_url_origin(),
+            redirect_count: 0,
+            destination: Destination::None,
+            initiator: Initiator::None,
+            nonce: "".to_owned(),
+            integrity_metadata: "".to_owned(),
+            parser_metadata: ParserMetadata::None,
+        };
+        let (result, violations) =
+            csp_list.should_navigation_request_be_blocked(&request, NavigationCheckType::Other);
```

Comment:
Can we get a TODO to populate this correctly, since navigation can be triggered by a form submission?